### PR TITLE
Quaternion to scaled-axis angle representation (and back) helpers

### PR DESCRIPTION
**Pull request by amzn-jillich**
Direct conversion helpers for quaternion to the scaled axis-angle representation and back without the need to convert them first to the axis-angle format and manually scale (or normalize on the way back). This also avoids having to deal with the special case of an identity representation which is 0,0,0 in the scaled axis-angle format while our convention for axis-angle is 0,1,0 for the axis and 0 for the angle.

Added unit tests that check the conversion round-trips from quaternion -> (scaled) axis-angle -> quaternion as well as comparing the scaled axis-angle representations from the direct helper functions as well as the axis-angle while manually scaling/normalizing.

Signed-off-by: Benjamin Jillich <jillich@amazon.com>

### CODE REVIEW

File: Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp [hunk 1]
```diff
@@ -408,4 +408,98 @@ namespace UnitTest
...
+            }
+
+            const AZ::Quaternion backFromAxisAngle = AZ::Quaternion::CreateFromAxisAngle(axisFromScaledResult, angleFromScaledResult);
+            EXPECT_TRUE(testQuat.IsClose(backFromAxisAngle, 1e-6f));
+        }
+    }
+
+    void TestQuaternionScaledAxisAngleRoundtripRotationCombinations(float angle)
+    {
+        TestQuaternionScaledAxisAngleRoundtrip(AZ::Quaternion::CreateRotationX(angle));
```
Comment (greerdv) [inline Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp:455]: This is a lot of checks in one test again. If this ever fails it will be tricky to track down exactly where it's failing. I think it might also be a bit overkill in terms of the number of checks. Personally I think it would be sufficient to call the Roundtrip test on just a few carefully chosen quaternions.

**Review by greerdv — APPROVED** [review]

**Review by lsemp3d — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Math/Quaternion.inl [hunk 2]
```diff
@@ -109,6 +109,25 @@ namespace AZ
...
+
+        if (halfAngle < AZ::Constants::FloatEpsilon)
+        {
+            return AZ::Quaternion::CreateFromVector3AndValue(exponentialMap, 1.0f).GetNormalized();
+        }
+        else
+        {
+            float sin, cos;
+            SinCos(halfAngle, sin, cos);
+            sin /= halfAngle;
```
[the remark this answers is not in the document]
Comment (amzn-jillich) [reply Code/Framework/AzCore/AzCore/Math/Quaternion.inl:125]: Changed. I went for (sin / halfAngle) * exponentialMap to not end up with two vector operations.

File: Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp [hunk 3]
```diff
@@ -408,4 +408,98 @@ namespace UnitTest
...
+        const AZ::Vector3 scaledAxisAngle = testQuat.ConvertToScaledAxisAngle();
+        {
+            // Convert the scaled axis-angle back into a quaternion.
+            AZ::Quaternion backFromScaledAxisAngle = AZ::Quaternion::CreateFromScaledAxisAngle(scaledAxisAngle);
+
+            // Compare the original quaternion with the one after the conversion to the scaled axis-angle representation.
+            EXPECT_TRUE(testQuat.IsClose(backFromScaledAxisAngle, 1e-6f));
+        }
+
+        // Axis-angle
```
[the remark this answers is not in the document]
Comment (amzn-jillich) [reply Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp:431]: Splitted them up.

File: Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp [hunk 4]
```diff
@@ -408,4 +408,98 @@ namespace UnitTest
...
+        TestQuaternionScaledAxisAngleRoundtrip(AZ::Quaternion::CreateRotationY(angle) * AZ::Quaternion::CreateRotationX(angle) * AZ::Quaternion::CreateRotationZ(angle));
+        TestQuaternionScaledAxisAngleRoundtrip(AZ::Quaternion::CreateRotationY(angle) * AZ::Quaternion::CreateRotationZ(angle) * AZ::Quaternion::CreateRotationX(angle));
+
+        TestQuaternionScaledAxisAngleRoundtrip(AZ::Quaternion::CreateRotationZ(angle) * AZ::Quaternion::CreateRotationX(angle) * AZ::Quaternion::CreateRotationY(angle));
+        TestQuaternionScaledAxisAngleRoundtrip(AZ::Quaternion::CreateRotationZ(angle) * AZ::Quaternion::CreateRotationY(angle) * AZ::Quaternion::CreateRotationX(angle));
+    }
+
+    TEST(MATH_Quaternion, ScaledAxisAngleConversionTests)
+    {
+        std::vector<float> specialCaseAngles
```
[the remark this answers is not in the document]
Comment (amzn-jillich) [reply Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp]: Splitted them up into several parameterized tests.

File: Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp (hunk 1 above)
Comment (amzn-jillich) [reply Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp:455]: Changed into less, more carefully chosen edge cases for the parameterized tests.

**Review by lsemp3d — APPROVED** [review]
